File: examples/java-repo/src/main/java/com/example/ecommerce/domain/order/Order.java
package com.example.ecommerce.domain.order;

import java.util.ArrayList;
import java.util.List;

public class Order {

    private final long id;
    private final long customerId;
    private final List<OrderLine> lines;
    private OrderStatus status;

    public Order(long id, long customerId) {
        if (id <= 0) {
            throw new IllegalArgumentException("order id must be positive");
        }
        if (customerId <= 0) {
            throw new IllegalArgumentException("customer id must be positive");
        }
        this.id = id;
        this.customerId = customerId;
        this.lines = new ArrayList<>();
        this.status = OrderStatus.PENDING;
    }

    public long getId() {
        return id;
    }

    public long getCustomerId() {
        return customerId;
    }

    // DELIBERATE: returns mutable internal list (SpotBugs EI_EXPOSE_REP)
    public List<OrderLine> getLines() {
        return lines;
    }

    public OrderStatus getStatus() {
        return status;
    }

    public void addLine(OrderLine line) {
        if (status != OrderStatus.PENDING) {
            throw new IllegalStateException("cannot add lines to a non-pending order");
        }
        lines.add(line);
    }

    public Money total() {
        if (lines.isEmpty()) {
            return Money.zero("USD");
        }
        Money sum = Money.zero(lines.get(0).getUnitPrice().getCurrency().getCurrencyCode());
        for (OrderLine line : lines) {
            sum = sum.add(line.lineTotal());
        }
        return sum;
    }

    public void confirm() {
        if (status != OrderStatus.PENDING) {
            throw new IllegalStateException("only pending orders can be confirmed");
        }
        this.status = OrderStatus.CONFIRMED;
    }

    public void markPaid() {
        if (status != OrderStatus.CONFIRMED) {
            throw new IllegalStateException("only confirmed orders can be marked as paid");
        }
        this.status = OrderStatus.PAID;
    }

    public void cancel() {
        if (status == OrderStatus.SHIPPED) {
            throw new IllegalStateException("shipped orders cannot be cancelled");
        }
        this.status = OrderStatus.CANCELLED;
    }

    // DELIBERATE: equals without hashCode (SpotBugs HE_EQUALS_USE_HASHCODE)
    @Override
    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (!(obj instanceof Order)) return false;
        Order other = (Order) obj;
        return this.id == other.id;
    }
}

File: examples/python-repo/ecommerce/domain/order/order.py
from ecommerce.domain.order import order_status
from ecommerce.domain.order.money import Money


class Order:
    def __init__(self, order_id, customer_id):
        if order_id <= 0:
            raise ValueError("order id must be positive")
        if customer_id <= 0:
            raise ValueError("customer id must be positive")

        self._id = order_id
        self._customer_id = customer_id
        self._lines = []
        self._status = order_status.PENDING

    @property
    def id(self):
        return self._id

    @property
    def customer_id(self):
        return self._customer_id

    @property
    def lines(self):
        return self._lines

    @property
    def status(self):
        return self._status

    def add_line(self, line):
        if self._status != order_status.PENDING:
            raise RuntimeError("cannot add lines to a non-pending order")
        self._lines.append(line)

    def total(self):
        if not self._lines:
            return Money.zero("USD")
        result = Money.zero(self._lines[0].unit_price.currency)
        for line in self._lines:
            result = result.add(line.line_total())
        return result

    def confirm(self):
        if self._status != order_status.PENDING:
            raise RuntimeError("only pending orders can be confirmed")
        self._status = order_status.CONFIRMED

    def mark_paid(self):
        if self._status != order_status.CONFIRMED:
            raise RuntimeError("only confirmed orders can be marked as paid")
        self._status = order_status.PAID

    def cancel(self):
        if self._status == order_status.SHIPPED:
            raise RuntimeError("shipped orders cannot be cancelled")
        self._status = order_status.CANCELLED

    # DELIBERATE: bare except (Ruff E722)
    def safe_total_str(self):
        try:
            return str(self.total())
        except:
            return "0 USD"
